### DISCUSSION TITLE
Flutter issues in Iran

### DISCUSSION THREAD

**Discussion by basirkhan1995** [community/community#122129] [Programming Help]
### Body

Hello friends!
I recently moved to Iran, due to high level of filtering Flutter is services is filtered in Iran, I have used DNS like shecan.ir and begzar.ir, in the early it was working a little bit, but not it is not working even using a VPN, I own v2ArrayN vpn, openVpn. when I open filtered website using chrome it works fine but not accessable using Android Studio VSCode and always show this error message. 
"Insufficient permissions to the resource at the https://pub.dev package repository". 
If any solution please help me. 
Thanks

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by charlestanev** [comment]
Hi! I can understand the frustration you're facing with restricted access to Flutter services in Iran. It sounds like you've already tried several solutions, including DNS and VPNs, but are still encountering the "Insufficient permissions" error in Android Studio and VSCode.

One possible solution might be setting up a custom proxy in Android Studio or VSCode to route traffic through a working VPN or other bypass methods. Have you tried tweaking the network configuration settings in these development environments? Sometimes, they have built-in options for proxy settings that might help bypass restrictions.

Another approach could be using an external server or a cloud-based development environment where you can code remotely, thus avoiding local restrictions. Services like GitHub Codespaces or Replit could be a workaround.

I'm curious if anyone else has faced similar issues in Iran or other regions with heavy filtering. If you have other recommendations, I'd be happy to hear them.

**Reply by ebndev** [reply]
Hi @charlestanev, 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by basirkhan1995** [comment]
Thanks for your reply, I really appreciate it.

I don't how to do it, if you can elaborate.

Thanks

**Comment by Amir-HShahi** [comment]
You can use [403 dns](https://403.online/download), for flutter in Iran.
if you want to use vpn instead of dns, windscribe is a good option.
some of v2ray configs works fine, but not all of them.
